aws/structure.go

```diff
@@ -4474,6 +4531,10 @@ func expandDynamoDbProjection(data map[string]interface{}) *dynamodb.Projection
 		projection.NonKeyAttributes = expandStringList(v)
 	}
 
+	if v, ok := data["non_key_attributes"].(*schema.Set); ok && v.Len() > 0 {
+		projection.NonKeyAttributes = expandStringList(v.List())
+	}
+
```

Comment:
since moving forward we'll expect the "non_key_attributes" to be a Set when extracted from the resource data, it'll be safe to remove the list handling above in L4530-4532